File: backend/app/Models/Package.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\SoftDeletes;
use Illuminate\Support\Str;

class Package extends Model
{
    use HasFactory, SoftDeletes;

    /**
     * The attributes that are mass assignable.
     *
     * @var array
     */
    protected $fillable = [
        'name',
        'slug',
        'description',
        'price',
        'duration_hours',
        'photo_count',
        'edited_photo_count',
        'include_makeup',
        'include_outfit',
        'features',
        'image',
        'is_active',
        'sort_order',
    ];

    /**
     * The attributes that should be cast.
     *
     * @var array
     */
    protected $casts = [
        'price' => 'decimal:2',
        'include_makeup' => 'boolean',
        'include_outfit' => 'boolean',
        'features' => 'array',
        'is_active' => 'boolean',
    ];

    /**
     * Boot the model
     */
    protected static function boot()
    {
        parent::boot();

        static::creating(function ($package) {
            if (empty($package->slug)) {
                $package->slug = Str::slug($package->name);
            }
        });

        // Hapus gambar saat paket dihapus
        static::deleting(function ($package) {
            if ($package->image) {
                \Illuminate\Support\Facades\Storage::disk('public')->delete($package->image);
            }
        });
    }

    /**
     * Relasi ke Bookings
     */
    public function bookings()
    {
        return $this->hasMany(Booking::class);
    }

    /**
     * Relasi ke Galleries
     */
    public function galleries()
    {
        return $this->hasMany(Gallery::class);
    }

    /**
     * Scope untuk filter active packages
     */
    public function scopeActive($query)
    {
        return $query->where('is_active', true);
    }

    /**
     * Scope untuk sort by order
     */
    public function scopeOrdered($query)
    {
        return $query->orderBy('sort_order', 'asc')->orderBy('name', 'asc');
    }

    /**
     * Get formatted price
     */
    public function getFormattedPriceAttribute(): string
    {
        return 'Rp' . number_format($this->price, 0, ',', '.');
    }

    /**
     * Get full image URL
     */
    public function getImageUrlAttribute()
    {
        return $this->image ? asset('storage/' . $this->image) : null;
    }
}

File: mobile/lib/models/package.dart
class Package {
  final int id;
  final String name;
  final String description;
  final double price;
  final int durationHours;
  final int photoCount;
  final int editedPhotoCount;
  final String? imageUrl;
  final bool isActive;

  Package({
    required this.id,
    required this.name,
    required this.description,
    required this.price,
    required this.durationHours,
    required this.photoCount,
    required this.editedPhotoCount,
    this.imageUrl,
    required this.isActive,
  });

  factory Package.fromJson(Map<String, dynamic> json) {
    return Package(
      id: json['id'],
      name: json['name'],
      description: json['description'],
      price: double.parse(json['price'].toString()),
      durationHours: json['duration_hours'],
      photoCount: json['photo_count'],
      editedPhotoCount: json['edited_photo_count'],
      imageUrl: json['image_url'],
      isActive: json['is_active'],
    );
  }
}
